### PR TITLE
Initial WIP code for user management

### PR DESCRIPTION
**Pull request by kburtram**
This PR contains some of the base functionality we'll need for User, Login and Role creation/update/delete.  This is early iteration of this feature area, so I'm not looking for extensive review, which we'll do separately as features reach more mature state.  Purpose of the PR is to move code into main branch so Hai and I can subdivide the work into our own branches.  Please do call out anything "broken" and I can address in this PR.

### CODE REVIEW

File: src/Directory.Build.props [hunk 1]
```diff
@@ -16,6 +16,8 @@
 		<RepositoryUrl>https://github.com/Microsoft/sqltoolsservice</RepositoryUrl>
 		<RepositoryType>git</RepositoryType>
 		<PackageTags>SQL XPLAT</PackageTags>
+		<NoWarn>$(NoWarn);CS8600;CS8601;CS8602;CS8603;CS8604;CS8605;CS8618;CS8620;CS8622;CS8625;CS8629;CS8632;CS8769;CS8765</NoWarn>
```
Comment (kburtram) [inline src/Directory.Build.props:19]: I wasn't intended for this in this PR.  I'll revert.

**Review by alanrenmsft — APPROVED** [review]
looks good to me, I have made changes to the interfaces/contracts based on our conversation this morning, but that shouldn't block this PR, you guys can always update it later.

**Review by caohai — APPROVED** [review]

File: src/Microsoft.SqlTools.ServiceLayer/Security/Contracts/UserInfo.cs [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        public bool isAAD { get; set; }
+
+        public ExtendedProperty[] ExtendedProperties { get; set; }
+
+        public SecurablePermissions[] SecurablePermissions { get; set; }   
+    }
+}
+
+
+#if false
```
Comment (alanrenmsft) [inline src/Microsoft.SqlTools.ServiceLayer/Security/Contracts/UserInfo.cs:80]: remove these typescript definitions

Comment (kburtram) [reply src/Microsoft.SqlTools.ServiceLayer/Security/Contracts/UserInfo.cs:80]: Will remove in next PR.

**Comment by kburtram** [comment]
Test failing in CI is also failing for other PRs and appears related to https://github.com/microsoft/sqltoolsservice/commit/9fe3aeddc31dbfc744d82a65f0a00ab18cce62b3.  I'll merge anyway since it's unrelated to this change.

**Pull request merged**